File: 70-ClimbingStairs/Solution.c
int climbStairs(int n) {
    if (n<=0) {
        return 0;
    }
    if (n==1) {
        return 1;
    }
    int res[3];
    res[1]=1;
    res[2]=2;
    for (short i=2; i<n; i++) {
        res[0]=res[1];
        res[1]=res[2];
        res[2]=res[1]+res[0];
    }
    return res[2];
}

File: 70-ClimbingStairs/Solution.ts
function climbStairs(n: number): number {
  const results: number[] = [];
  results[0]=0;
  results[1]=1;
  results[2]=2;
  results[3]=3;
  for (let i=4; i<=n; i++) {
      results[i]=results[i-1]+results[i-2];
  }
  return results[n];
};